workos/fga.py

```diff
@@ -0,0 +1,384 @@
+from typing import Any, Dict, List, Optional, Protocol
+
+import workos
+from workos.types.fga import (
+    CheckOperation,
+    CheckResponse,
+    Resource,
+    ResourceType,
+    Warrant,
+    WarrantCheck,
+    WarrantWrite,
+    WarrantWriteOperation,
+    WriteWarrantResponse,
+)
+from workos.types.fga.list_filters import ResourceListFilters, WarrantListFilters
+from workos.types.list_resource import (
+    ListArgs,
+    ListMetadata,
+    ListPage,
+    WorkOsListResource,
+)
+from workos.utils.http_client import SyncHTTPClient
+from workos.utils.pagination_order import Order, PaginationOrder
+from workos.utils.request_helper import (
+    REQUEST_METHOD_DELETE,
+    REQUEST_METHOD_GET,
+    REQUEST_METHOD_POST,
+    REQUEST_METHOD_PUT,
+    RequestHelper,
+)
+from workos.utils.validation import Module, validate_settings
+
+DEFAULT_RESPONSE_LIMIT = 10
+
+ResourceListResource = WorkOsListResource[Resource, ResourceListFilters, ListMetadata]
+
+ResourceTypeListResource = WorkOsListResource[Resource, ListArgs, ListMetadata]
+
+WarrantListResource = WorkOsListResource[Warrant, WarrantListFilters, ListMetadata]
+
+
+class FGAModule(Protocol):
+    def get_resource(self, resource_type: str, resource_id: str) -> Resource: ...
+
+    def list_resources(
+        self,
+        resource_type: Optional[str] = None,
+        search: Optional[str] = None,
+        limit: int = DEFAULT_RESPONSE_LIMIT,
+        order: PaginationOrder = Order.Desc.value,
```

Comment:
Made me realize we actually meant to remove Order. Could just use `PaginationOrder = "desc"` here.